site/app/controllers/forum/ForumController.php

```diff
@@ -1243,4 +1242,18 @@ public function showStats() {
         ksort($users);
         $this->core->getOutput()->renderOutput('forum\ForumThread', 'statPage', $users);
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/post/likes", methods={"POST"})
+     */
+    public function toggleLike() {
+        $output= [];
+        $output['type']=$this->core->getQueries()->toggleLikes($_POST['post_id'], $_POST['thread_id'], $_POST['current_user']);
```

Comment:
This will cause a server error if those keys don't exist. Could you make a small check before hand and return a json fail if they don't exist.